cpp/daal/src/algorithms/kmeans/kmeans_container.h

```diff
@@ -83,9 +85,11 @@ services::Status BatchContainer<algorithmFPType, method, cpu>::compute()
     Parameter * par                        = static_cast<Parameter *>(_par);
     daal::services::Environment::env & env = *_env;
 
+    oneapi::dal::network::empty_network net;
+
     if (deviceInfo.isCpu || method != lloydDense)
     {
-        __DAAL_CALL_KERNEL(env, internal::KMeansBatchKernel, __DAAL_KERNEL_ARGUMENTS(method, algorithmFPType), compute, a, r, par);
+        __DAAL_CALL_KERNEL(env, internal::KMeansBatchKernel, __DAAL_KERNEL_ARGUMENTS(method, algorithmFPType), compute, a, r, par, net);
```

Comment:
Maybe it make sense to take network from env like it's done for execution context? In that cases user will set proper network object to environment. By default, empty network will be set. In that case, there is no need to pass network as a parameter to kernel - kernel will be able to take it from environment